src/app/components/Transaction/InfoBox.tsx

```diff
@@ -1,23 +1,34 @@
 import { Box, Text } from 'grommet'
 import * as React from 'react'
+import type { Icon } from 'grommet-icons/icons'
+import copy from 'copy-to-clipboard'
+import { trimLongString } from '../ShortAddress'
 
-interface DetailProps {
-  icon?: React.ReactNode
+interface InfoBoxProps {
+  copyToClipboard?: boolean
+  icon: Icon
   label: string
-  value: string | React.ReactNode
+  value: string
+  trimValue?: boolean
 }
 
-export function InfoBox(props: DetailProps) {
+export function InfoBox({ copyToClipboard, icon: IconComponent, label, trimValue, value }: InfoBoxProps) {
   return (
-    <Box direction="row" gap="small" hoverIndicator={{ color: 'background-contrast' }} pad="medium">
-      {props.icon && (
-        <Box fill="vertical" align="center" justify="center" alignSelf="center" pad={{ right: 'xsmall' }}>
-          {props.icon}
-        </Box>
-      )}
+    <Box
+      direction="row"
+      gap="small"
+      hoverIndicator={{ color: 'background-contrast' }}
+      pad={{ horizontal: 'small', vertical: 'small' }}
+      onClick={copyToClipboard ? () => copy(value) : undefined}
+      style={{ boxShadow: 'none' }}
```

Comment:
This is terrible for keyboard accessibility

Reply:
hm indeed that sucks. will fix that tomorrow.